src/sid/update_states.py

```diff
@@ -202,34 +202,92 @@ def update_derived_state_variables(states, derived_state_variables):
 
 
 def _update_immunity_level(states: pd.DataFrame, params: pd.DataFrame) -> pd.DataFrame:
-    """Update immunity level from infection and vaccination."""
-    immunity = states["immunity"].copy()
+    """Update immunity levels from infection and vaccination."""
+    days_since_infection = states["cd_infectious_true"]
+    days_since_vaccination = states["cd_is_immune_by_vaccine"]
```

Comment:
I'm not sure if this equality holds, but if we have the variables on the left the following code should work fine.

Reply:
I think both are not right. 

For days since infection I would use `cd_ever_infected`. The `cd_infectious_true` counter only starts a few days after your infection. 

For the `cd_is_immune_by_vaccine` you would need to add the `cd_is_immune_by_vaccine_draws` column to get the days since the vaccination since it starts counting down from there when the person is vaccinated.

Reply:
Creating a countdown from the date of the vaccination was a better solution. :+1: